Lib/asyncore.py

```diff
@@ -127,7 +127,8 @@ def poll(timeout=0.0, map=None):
         map = socket_map
     if map:
         r = []; w = []; e = []
-        for fd, obj in list(map.items()):
+        map_copy = dict(map)
```

Comment:
Thank you for the review, 
If this is the case then we are misusing asyncore, however that would surprise me since I guess the `list` in `list(map.items())` is there precisely because `map` may change during the iteration.

Also, looks like there are [tests with threads](https://github.com/python/cpython/blob/master/Lib/test/test_asyncore.py):